src/codegen.cpp

```diff
@@ -2523,6 +2529,21 @@ static Value *emit_f_is(jl_codectx_t &ctx, const jl_cgval_t &arg1, const jl_cgva
     if (jl_type_intersection(rt1, rt2) == (jl_value_t*)jl_bottom_type) // types are disjoint (exhaustive test)
         return ConstantInt::get(T_int1, 0);
 
+    if ((arg1.V || arg1.constant) && (arg2.V || arg2.constant) &&
+        (rt1 == (jl_value_t*)jl_bool_type || rt2 == (jl_value_t*)jl_bool_type ||
+         rt1 == (jl_value_t*)jl_int8_type || rt2 == (jl_value_t*)jl_int8_type ||
+         rt1 == (jl_value_t*)jl_uint8_type || rt2 == (jl_value_t*)jl_uint8_type)) {
+        // Both sides are boxed or can be trivially boxed
+        // also at least one side has known stable boxed address,
+        // we can simply compare the pointer.
```

Comment:
Only jl_true and jl_false have stable addresses guaranteed. The others are allowed to get copies (for example, the system image and runtime intrinsics don't care if they make copies, and they'd also be even allowed to appear on the stack, for example)

Reply:
Stack should not be an issue, since this uses `boxed` that should always return the right thing. I'm pretty sure no one else is allocating them so the only issue is the system image storing extra copies of them? Seems like it might be enough to add the `Int8` and `UInt8` boxes to the `_tags` list and fix the initialization?